website/project/tasks.py

```diff
@@ -102,15 +104,18 @@ def format_node(node):
         }, {
             '@id': '_:789',
             '@type': 'project',
-            'is_deleted': not node.is_public or node.is_deleted or node.is_spammy,
+            'is_deleted': not node.is_public or node.is_deleted or node.is_spammy or is_qa_node
         }
     ]
 
 def format_registration(node):
+    is_qa_node = bool(set(settings.DO_NOT_INDEX_LIST['tags']).intersection(node.tags.all().values_list('name', flat=True))) \
+        or any(substring in node.title for substring in settings.DO_NOT_INDEX_LIST['titles'])
+
     registration_graph = GraphNode('registration', **{
         'title': node.title,
         'description': node.description or '',
-        'is_deleted': not node.is_public or 'qatest' in (node.tags.all() or []) or node.is_deleted,
+        'is_deleted': not node.is_public or is_qa_node,
```

Comment:
You seem to have gotten rid of the `node.is_deleted` case in this statement, which I _suspect_ is important, unless it's happening implicitly somehow in `is_qa_node`, but if so, it's pretty subtle and might need to be made more explicit.

Reply:
You are definitely right....... I somehow removed the `is_deleted` when I added the `is_qa_node`.

Reply:
I'm a little concerned that it didn't break anything, and I expect it's because it's in a task. We'll probably need to fix that at some point, but not for this as long as you run some manual tests and do your best to make sure it's okay before it goes to QA.

Reply:
Since this is formatting the payload of request to SHARE for updating node information on SHARE side, we can't really test it with OSF unit tests. I also can't do manual tests that confirm whether the change would remove the item from SHARE results. However I did confirm with Abram that if we we want to remove a node from SHARE we only need to set the `is_deleted` to True in the payload. In this case, I am simply adding another condition to set the `is_deleted` field so this should be fine.

Reply:
Umm... On second thought, we could at least write some tests to make sure the payload it formatted properly upon node updates. It seems currently we do have some tests doing some trivial test on the payload (e.g. whether the payload is formatted correctly) but nothing substantial.